### PR TITLE
Update endpoint for autoscaler options for KEB

### PR DESCRIPTION
**Pull request by wozniakjan**
**Description**

Enhance KEB instance update endpoint to allow users to modify autoscaler parameters

Sample test using port-forwarded deployment:
```
instance=test-jw34
curl -XPATCH "localhost:8080/oauth/v2/service_instances/$instance?accepts_incomplete=true" -i \
    -H "X-Broker-API-Version: 2.14" \
    -H "Content-Type: application/json" \
    --data-binary @- << EOF
{
   "service_id":"47c9dcbf-ff30-448e-ab36-d3bad66ba281",
   "plan_id":"4deee563-e5ec-4731-b9b1-53b42d855f0c",
   "context":{
       "globalaccount_id":"***",
       "subaccount_id":"***",
       "user_id":"jan.wozniak@sap.com"
   },
   "parameters":{
       "name":"$instance",
       "autoScalerMin": 10,
       "autoScalerMax": 11,
       "maxSurge": 12,
       "maxUnavailable": 13
   }
}
EOF
```

and then connecting to the broker SQL database
```
broker=> select type, finished_stages, provisioning_parameters from operations where instance_id='test-jw34' order by created_at desc limit 5;
   type    |    finished_stages    |   provisioning_parameters
-----------+-----------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 update    |                       | {"plan_id":"4deee563-e5ec-4731-b9b1-53b42d855f0c","service_id":"47c9dcbf-ff30-448e-ab36-d3bad66ba281","ers_context":{"tenant_id":"","subaccount_id":"***","globalaccount_id":"***","active":true,"user_id":"jan.wozniak@sap.com"},"parameters":{"name":"test-jw34","targetSecret":null,"volumeSizeGb":null,"machineType":null,"region":null,"purpose":null,"licence_type":null,"zones":null,"zonesCount":null,"autoScalerMin":10,"autoScalerMax":11,"maxSurge":12,"maxUnavailable":13,"components":null,"kymaVersion":"","overridesVersion":"","administrators":null,"provider":null},"platform_region":"cf-eu10","platform_provider":"unknown"}
 provision | ,start,create_runtime | {"plan_id":"4deee563-e5ec-4731-b9b1-53b42d855f0c","service_id":"47c9dcbf-ff30-448e-ab36-d3bad66ba281","ers_context":{"tenant_id":"","subaccount_id":"***","globalaccount_id":"***","user_id":"jan.wozniak@sap.com"},"parameters":{"name":"test-jw34","targetSecret":"sap-skr-dev-cust-00002-kyma-integration","volumeSizeGb":null,"machineType":null,"region":null,"purpose":null,"licence_type":null,"zones":null,"zonesCount":null,"autoScalerMin":5,"autoScalerMax":7,"maxSurge":3,"maxUnavailable":4,"components":null,"kymaVersion":"","overridesVersion":"","administrators":null,"provider":null},"platform_region":"cf-eu10","platform_provider":"unknown"}
```
in the `provisioning_parameters`, the values were updated:
* `autoScalerMin` 5 => 11
* `autoScalerMax` 7 => 12
* `maxSurge` 3 => 13
* `maxUnavailable` 4 => 14

**Related issue(s)**

### CODE REVIEW

**Comment by wozniakjan** [comment]
mind that you may need to update the image tag if you want this change to be deployed. It's still manual labour :)

https://github.com/kyma-project/control-plane/blob/9126bdcd65aeb3eef464cd24f025e83faec23557/resources/kcp/values.yaml#L22-L24

File: components/kyma-environment-broker/internal/dto.go [hunk 1]
```diff
@@ -102,6 +102,31 @@ type ProvisioningParametersDTO struct {
 type UpdatingParametersDTO struct {
 	OIDC                  *OIDCConfigDTO `json:"oidc,omitempty"`
 	RuntimeAdministrators []string       `json:"administrators,omitempty"`
+	AutoScalerMin         *int           `json:"autoScalerMin"`
+	AutoScalerMax         *int           `json:"autoScalerMax"`
+	MaxSurge              *int           `json:"maxSurge"`
+	MaxUnavailable        *int           `json:"maxUnavailable"`
+}
+
+func (u UpdatingParametersDTO) UpdateAutoScaler(p *ProvisioningParametersDTO) bool {
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/dto.go:111]: nit: do we need the returned bool value? looks like it is not used

Comment (wozniakjan) [reply components/kyma-environment-broker/internal/dto.go:111]: It is actually used here:
https://github.com/kyma-project/control-plane/blob/393deaf9b4073b0c5e03c6727095f150d2e7952f/components/kyma-environment-broker/internal/broker/instance_update.go#L173-L175

Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/dto.go:111]: yes, but the returned bool value is not used

Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/dto.go:111]: are you planning to use it?

Comment (wozniakjan) [reply components/kyma-environment-broker/internal/dto.go:111]: it is used in the if statement I pasted above. If the functions returns true, it appends to `updateStorage`, else it skips the append

**Comment by piotrmiskiewicz** [comment]
in general it looks nice, the TestUpdateAutoscalerParams test is saying that it works :)

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**